### DISCUSSION TITLE
关于逐字歌词以及歌词编辑

### DISCUSSION THREAD

**Discussion by lrisora** [zhongyang219/MusicPlayer2#397] [General]
支持多种格式的编辑没必要，只支持读取就足够了
我建议逐字歌词进入编辑状态全部采用扩展lrc格式

也就是遇到其他格式的歌词一旦发生编辑（包括调整偏移量）就立即另存为转换为扩展lrc歌词
或者只要打开就进行转换

有一点牺牲，扩展lrc无法保存行结束时间，不过不是大问题
扩展lrc优势就是给人读直观，编辑方便

**Comment by lifegpc** [comment]
行结束时间其实靠插入空行就解决了

**Reply by lrisora** [reply]
在目前的代码实现里，这两种看起来是不同的，插入空行会使多行歌词显示出一个实在的空行
而最后一个字时间结束后行结束时间还没到会在当前行停留一下

不建议使用空行代换这个时间差，直接不管看起来效果最好，会显示下一句歌词
只是不确定会不会有故意要当前行结束后非要停留的需求
我记得现有代码只要
[01:05.37]ラ[01:06.39]ラ[01:07.45]ラ[01:09.13][01:12.13]
就会停到01:12.13，相当于加了一个不存在的字
由于不清楚扩展lrc的标准，不知道与其他软件的兼容性怎么样

不过我还没见过最后一个字时间与行结束时间故意相差能看出来的时间的歌词
这个问题可能本就不存在，不过保险起见

**Reply by lifegpc** [reply]
> 在目前的代码实现里，这两种看起来是不同的，插入空行会使多行歌词显示出一个实在的空行 而最后一个字时间结束后行结束时间还没到会在当前行停留一下

也许可以实现一下，间隔不超过一个指定值就隐藏掉
> 不建议使用空行代换这个时间差，直接不管看起来效果最好，会显示下一句歌词 只是不确定会不会有故意要当前行结束后非要停留的需求 我记得现有代码只要 [01:05.37]ラ[01:06.39]ラ[01:07.45]ラ[01:09.13][01:12.13] 就会停到01:12.13，相当于加了一个不存在的字 由于不清楚扩展lrc的标准，不知道与其他软件的兼容性怎么样

基本没见过
另外现在的翻译歌词格式兼容也很差（为了让别的软件能兼容，我都是自己写脚本转换的

**Reply by lifegpc** [reply]
![image](https://user-images.githubusercontent.com/41434272/183292769-9298f5e3-73cd-4103-882d-0ca7230b9dc5.png)
来首一行空行一行歌词的歌

**Comment by lifegpc** [comment]
![image](https://user-images.githubusercontent.com/41434272/184275039-8a087d19-d993-400d-a12e-2a7c70f73f7c.png)
总感觉怪怪的，不知道是源的问题还是处理时候的问题

**Reply by lifegpc** [reply]
https://music.163.com/api/song/lyric?os=osx&id=36496753&lv=-1&kv=-1&tv=-1
看上去是处理的问题
[00:42.41] 那行 API 里是全部都有的
```lrc
[00:39.04]あれこれそれもしよっ[00:42.41]ぴゅぴゅーんって飛び出した（飛び出した）\n
[00:42.41]Piu Piu地跳起来\n
```
看上去是[00:42.41]前少了个\n
@lrisora

**Reply by lifegpc** [reply]
想了想好像也没有什么特别好的办法。也许可以加个flag，从网易云下歌词的时候，把同一行新的时间视为新的一行歌词？
